lib/sles4sap/azure_cli.pm

```diff
@@ -1374,3 +1378,163 @@ sub az_resource_delete {
 
     assert_script_run(join(' ', @az_command), timeout => $args{timeout});
 }
+
+=head2 az_storage_blob_upload
+
+    az_storage_blob_upload(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount',
+        file=>'somefilename' [, timeout=>42]);
+
+Uploads file to a storage container.
+
+=over 4
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<file> File to upload.
+
+=item B<timeout> Timeout for az command.
+
+=back
+=cut
+
+sub az_storage_blob_upload {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name', 'file') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{timeout} //= '90';
+
+    my $az_cmd = join(' ',
+        'az storage blob upload',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--file $args{file}"
+    );
+    assert_script_run(join(' ', $az_cmd), timeout => $args{timeout});
+}
+
+=head2 az_storage_blob_lease_acquire
+
+    az_storage_blob_lease_acquire(
+        container_name=>'somecontainer',
+        storage_account_name=>'storageaccount',
+        blob_name => 'somefilename' [, lease_duration=>'42']
+    );
+
+Acquire a lease for a storage blob.
+
+=over 4
+
+=item B<container_name> Existing storage container name.
+
+=item B<storage_account_name> Storage account name.
+
+=item B<blob_name> Blob name to acquire lease for.
+
+=item B<lease_duration> Lease duration between 15-60s. Default: infinite
+
+=back
+=cut
+
+sub az_storage_blob_lease_acquire {
+    my (%args) = @_;
+    foreach ('container_name', 'storage_account_name', 'blob_name') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    $args{lease_duration} //= '-1';    # -1 = infinite lease
+
+    my $az_cmd = join(' ',
+        'az storage blob lease acquire',
+        '--only-show-errors',
+        "--container-name $args{container_name}",
+        "--account-name $args{storage_account_name}",
+        "--blob-name $args{blob_name}",
+        "--lease-duration $args{lease_duration}",
+        '--output tsv'    # Json output won't work here.
```

Comment:
could you please explain and annotate why not?

Reply:
If az cli command fails to acquire a lease to the file, it will just return an error message which is not a valid json format. If you pass this through `decode_json()` as we usually do, the test will die with `malformed json format` error. 
The purpose of this function is to either return UUID or nothing instead of failing.